tests/foreman/api/test_webhook.py

```diff
@@ -0,0 +1,173 @@
+"""Test class for Webhook API
+
+:Requirement: Webhook
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: Webhooks
+
+:Assignee: gsulliva
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import re
+
+import pytest
+from nailgun import entities
+from requests.exceptions import HTTPError
+from wait_for import TimedOutError
+from wait_for import wait_for
+
+from robottelo.config import settings
+from robottelo.constants import WEBHOOK_EVENTS
+from robottelo.constants import WEBHOOK_METHODS
+from robottelo.datafactory import parametrized
+from robottelo.logging import logger
+
+
+def _read_log(ch, pattern):
+    """Read the first line from the given channel buffer and return the matching line"""
+    # read lines until the buffer is empty
+    for log_line in ch.stdout().splitlines():
+        logger.debug(f'foreman-tail: {log_line}')
+        if re.search(pattern, log_line):
+            return log_line
+    else:
+        return None
+
+
+def _wait_for_log(channel, pattern, timeout=2, delay=0.2):
+    """_read_log method enclosed in wait_for method"""
+    matching_log = wait_for(
+        _read_log,
+        func_args=(
+            channel,
+            pattern,
+        ),
+        fail_condition=None,
+        timeout=timeout,
+        delay=delay,
+        logger=logger,
+    )
+    return matching_log.out
+
+
+def assert_event_triggered(channel, event):
+    """Reads foreman logs until event trigger message is found"""
+    pattern = f'ForemanWebhooks::EventSubscriber: {event} event received'
+    try:
+        log = _wait_for_log(channel, pattern)
+        assert pattern in log
+    except TimedOutError:
+        raise AssertionError(f'Timed out waiting for {pattern} from VM')
+
+
+class TestWebhook:
+    @pytest.mark.tier2
+    def test_negative_invalid_event(self):
+        """Test negative webhook creation with an invalid event
+
+        :id: 60cd456a-9943-45cb-a72e-23a83a691499
+
+        :expectedresults: Webhook is not created
+
+        :CaseImportance: High
+        """
+        with pytest.raises(HTTPError):
+            entities.Webhooks(event='invalid_event').create()
```

Comment:
Instead of calling with entities, I would suggest using target_sat/class_target_sat

Reply:
Why is that?

Reply:
It's good to have, but it's not strictly necessary